pkgs/top-level/all-packages.nix

```diff
@@ -31876,6 +31876,15 @@ with pkgs;
   minetestclient = minetestclient_5;
   minetestserver = minetestserver_5;
 
+  minetest-mods = callPackage ../games/minetest/mods { };
+
+  minetestWithMods = import ../games/minetest/minetest-with-mods.nix
+                                pkgs
+                                {
+                                    inherit minetest-mods;
+                                    inherit minetest;
+                                  };
```

Comment:
Do we want this to work with games and textures too? If yes, should we `s/mods/content/`?

Also you mention generating the package set from ContentDB. This is another think I also want to do, but we should plan ahead the attrset hierarchy. Technical names in ContentDB are unique **except for forks** so we probably want to namespace content by user. I propose:

```nix
minetestContent = {
  user1 = {
    game1 = ...;
    mod1 = ...;
    texture1 = ...;
  };
  user2 = {
    game2 = ...;
    mod1 = ...; # fork
  };
  extraContent = {
    # Stuff not from ContentDB goes here.
  };
};
```

Reply:
Hmm, maybe putting extraContent in the ContentDB namespace is a bad idea (what happens if a "extraContent" user registers?).

We could do it like this

```nix
minetestContent = {
  contentDB = {
    user1 = {
      game1 = ...;
      mod1 = ...;
      texture1 = ...;
    };
    user2 = {
      game2 = ...;
      mod1 = ...; # fork
    };
  };
  extraContent = {
    # Stuff not from ContentDB goes here.
  };
};
```

...which results in attrpaths a bit too long for my liking (`minetestContent.contentDb.user.mod`). There's precedent though, such as `haskell.packages.ghcXYZ.pkg`

Reply:
Having this for all content would definitely be nice, I'm not sure whether it would be good to integrate that into this PR. I should be no problem to deal with that separately and then just override the `minetest` argument to combine the two package modifications.

Regarding your solution to the name space problem, I think thats how we should do it.

Reply:
Makes sense, but looking forward to that I'd still call the top-level attribute minetestContent or minetest-content